kubernetes/util.go

```diff
@@ -132,6 +132,14 @@ func DiscoverKubernetesNode(log *logp.Logger, nd *DiscoverKubernetesNodeParams)
 		log.Debug(err)
 	}
 
+	// use environment variable NODE_NAME
+	node := os.Getenv("NODE_NAME")
+	if node != "" {
```

Comment:
Any tests to be updated or added here? I see we have some at https://github.com/elastic/elastic-agent-autodiscover/blob/6deaacb6f7fdded3e61ed77358170cdca674b87f/kubernetes/util_test.go#L37 but I wonder if those are up to date.

Reply:
There are test for every use case. But the tests do not test if the machine-id provided is actually correct one. This would require api calls. The tests test the order of the steps execution for every case. I added another one to resemble the use case of wrong machine-id.